lib/compat.php

```diff
@@ -131,10 +131,22 @@ function gutenberg_ensure_wp_api_request() {
  * @return array           Filtered settings.
  */
 function gutenberg_disable_editor_settings_wpautop( $settings ) {
-	if ( ! isset( $settings['wpautop'] ) ) {
+	if ( is_array( $settings ) ) {
 		$settings['wpautop'] = ! gutenberg_post_has_blocks( get_post()->ID );
 	}
 
 	return $settings;
 }
 add_filter( 'wp_editor_settings', 'gutenberg_disable_editor_settings_wpautop' );
+
+/**
+ * Jetpack Markdown support causes issues when saving a Gutenberg post
+ * by stripping out the <p> tags - this is a bit heavy handed approach
+ * but disabling does fix the issue
+ */
+function gutenberg_remove_jetpack_markdown_support() {
+	if ( defined( 'JETPACK__VERSION' ) ) {
```

Comment:
Would it hurt if we removed `wpcom-markdown` support without explicitly testing whether we're in the context of Jetpack, since I assume the post support (by its prefix) is relevant for Jetpack only anyways?

Reply:
I changed the method of determining and removed it regardless of JETPACK being defined.